lib/matplotlib/cbook/__init__.py

```diff
@@ -1045,28 +1028,14 @@ def __iter__(self):
         The iterator is invalid if interleaved with calls to join().
         """
         self.clean()
-        token = object()
-
-        # Mark each group as we come across if by appending a token,
-        # and don't yield it twice
-        for group in self._mapping.values():
-            if group[-1] is not token:
-                yield [x() for x in group]
-                group.append(token)
-
-        # Cleanup the tokens
-        for group in self._mapping.values():
-            if group[-1] is token:
-                del group[-1]
+        for group in ({id(group): group for group in self._mapping.values()}
```

Comment:
I can understand what the other code is doing, though I agree it looks inelegant.  This change is maybe  more elegant, but its not very clear whats happening.  Can you add a line of explanation for the folks who don't parse more obscure python constructs?

Reply:
An additional variable may help:
~~~
unique_groups = {id(group): group for group in self._mapping.values()}.values()
for group in unique_groups:
~~~